File: SWEA/D3/1225. ［S／W 문제해결 기본］ 7일차 － 암호생성기/［S／W 문제해결 기본］ 7일차 － 암호생성기.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

class Solution
{
    public static void main(String args[]) throws Exception
    {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        
        for (int t = 0; t < 10; t++) {
            int test_case = sc.nextInt();
            Queue<Integer> q = new LinkedList<>();
            
            for (int j = 0; j < 8; j++) { 
                q.offer(sc.nextInt());
            }
            
            int count = 1;
            while (true) {
                int front = q.poll();
                front -= count;
                
                if (front <= 0) {
                    q.offer(0);
                    break; 
                }
                
                q.offer(front);
                
                count++;
                if (count > 5) {
                    count = 1;
                }    
            }
            
            sb.append("#").append(test_case).append(" ");
            for (int num : q) {
                sb.append(num).append(" ");
            }
            sb.append("\n");
        }
       
        System.out.print(sb.toString());
    }
}

File: SWEA/D3/1225. ［S／W 문제해결 기본］ 7일차 － 암호생성기/［S／W 문제해결 기본］ 7일차 － 암호생성기.py
from collections import deque

for _ in range(10):
    test_case = int(input())
    q = deque(list(map(int, input().split())))

    count = 1
    while True:
        if count == 6:
            count = 1

        front = q.popleft()

        if front - count <= 0:
            q.append(0)
            break

        q.append(front - count)
        count += 1

    print(f"#{test_case} ", end="")
    for n in q:
        print(f"{n} ", end="")
    print()